sky/serve/autoscalers.py

```diff
@@ -268,26 +271,20 @@ def _select_replicas_to_scale_down(
             cls, num_limit: int,
             replica_infos: Iterable['replica_managers.ReplicaInfo']
     ) -> List[int]:
-
         status_order = serve_state.ReplicaStatus.scale_down_decision_order()
-        # Also sort by provisioned time (indicated by replica_id) to make sure
-        # we terminate the replicas that starts provisioning later first
-        replica_infos_sorted = sorted(
+        replicas = sorted(
             replica_infos,
             key=lambda info: (
                 status_order.index(info.status)
                 # Use -1 for other status that will be terminated first.
                 # Including: NOT_READY, SHUTTING_DOWN, FAILED,
                 # FAILED_CLEANUP, PREEMPTED, UNKNOWN.
                 if info.status in status_order else -1,
-                # `-info.replica_id` is to furhter sort the replicas with the
-                # same state by the time it starts launching. In that case,
-                # if two replicas are both in PROVISIONING state, we will scale
-                # down the one that starts provisioning later, i.e., it will
-                # take a longer time for it to be READY.
-                -info.replica_id))
+                # `info.replica_id` is to furhter sort the replicas so that the
+                # older (version) replicas are scaled down first.
+                info.replica_id))
```

Comment:
```suggestion
                # `info.replica_id` is to furhter sort the replicas so that the
                # older (version) replicas are scaled down first.
                info.version, -info.replica_id))
```
should this use `info.version` first and then still use `-replica_id` to terminate the latter launched once first

Reply:
Good point! Fixed. Thanks!